sdks/python/apache_beam/transforms/core.py

```diff
@@ -320,6 +320,22 @@ def split_and_size(self, element, restriction):
     for part in self.split(element, restriction):
       yield part, self.restriction_size(element, part)
 
+  def truncate(self, element, restriction):
```

Comment:
Should we also provide truncate_and_size?

Reply:
`restriction_size` is a required API, thus `truncate` is enough here since the caller should be the system.